File: lexer/token.js
export default class Token {
    constructor(type_, value, pos_start, pos_end) {
        this.type = type_;
        this.value = value;
        this.pos_start = pos_start;
        this.pos_end = pos_end;
    }

    in(...args) {
        for (let r of args) {
            if (this.type === r) {
                return true;
            }
        }
        return false
    }

    matches(...args) {
        for (let i = 0; i < args.length; i++) {
            if (this.type === args[i][0] && (args[i].length > 1 ? (this.value === args[i][1]) : true)) {
                return true;
            }
        }
        return false;
    }

    toString() {
        if (this.value) return `[${this.type}:${this.value}]`;
        return `[${this.type}]`;
    }
}

File: python/token.py
class Token:
    def __init__(self, _type, value, pos_start, pos_end):
        self.type = _type
        self.value = value
        self.pos_start = pos_start
        self.pos_end = pos_end
    
    def __eq__(self, r):
        if isinstance(r, str):
            return self.type == r
        if not isinstance(r, Token):
            return False
        return self.type == r.type
    
    def __gt__(self, r):
        if isinstance(r, str):
            return self.value == r
        if not isinstance(r, Token):
            return False
        return self.value == r.value
    
    def __lt__(self, s):
        if not isinstance(s, str):
            return False
        return f'{self.type}:{self.value}' == s

    def __repr__(self):
        return f'<{self.type}:{self.value}>'